server/server.go

```diff
@@ -749,6 +757,13 @@ func (s *Server) copyInfo() Info {
 		info.ClientConnectURLs = make([]string, len(s.info.ClientConnectURLs))
 		copy(info.ClientConnectURLs, s.info.ClientConnectURLs)
 	}
+	if s.nonceRequired() {
+		// Nonce handling
+		var raw [nonceLen]byte
+		nonce := raw[:]
+		s.generateNonce(nonce)
+		info.Nonce = string(nonce)
```

Comment:
I think that we need to generate a new nonce from time to time. It could be after random number of client connects, regenerate a new nonce. Problem is then we need to know which nonce is "assigned" to a connect challenge.

Reply:
i think we need a new nonce on every challenge. Repeated nonces are the devils workshop.

My branch (jwt) has the nonce on the connection so we guarantee that the response is the right one, i think that is important because otherwise people can reply nonces, even if they aren't reused.

Reply:
The nonce is generated on a new connection and when we expect a CONNECT from the client. When we change the code to want new CONNECTS (proto change) we will make sure to generate a new one. I have tests that make sure they are different for new clients etc.